File: Easy/350. Intersection of Two Arrays II/main.cpp
#include <iostream>
#include <string>
#include <vector>
#include <unordered_map>


using namespace std;

class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {

        unordered_map<int, int> map;
        vector<int> inter;
        for (auto&& i : nums1) {
            map[i]++;
        }

        for (auto&& num : nums2) {
            if (map[num] >= 1) {
                inter.push_back(num);
                map[num]--;
            }
        }
        return inter;
    }
};



int main() {




    Solution s = Solution();
    vector<int> t1 = { 2,1 };
    vector<int> t2 = { 1,2 };

    auto res = s.intersect(t1, t2);

    for (auto&& i : res)
    {
        cout << i << endl;
    }



    return 0;
}

File: Easy/350. Intersection of Two Arrays II/main.py

class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        if len(nums2)>len(nums1):
            temp_nums=nums1
            nums1=nums2
            nums2=temp_nums
        dic={}
        for num in nums2:
            if num in dic:
                dic[num]+=1
            else:
                dic[num]=1
        intersecion_list=[]
        for i in nums1:
            if i in dic:
                if dic[i]>=1:
                    intersecion_list.append(i)
                    dic[i]-=1
        return intersecion_list
                    
s=Solution()

test1=[2,1]
test2=[1,2]

print(s.intersect(test1,test2))
